pkgs/pkgs-lib/formats.nix

```diff
@@ -473,4 +473,56 @@ rec {
     '') {};
   };
 
+  envvars = {
+    export ? false
+  , listToStringSep ? ","
+  , listToStringFn ? lib.concatStringsSep listToStringSep
+  , boolToStringFn ? lib.boolToString
+  }: {
+    type = with lib.types; let
+      unescapedType = addCheck (attrsOf str) (attrs: (lib.attrNames attrs) == [ "_type" "value" ] && attrs._type == "unescaped");
+      valueType = nullOr (oneOf [
+        bool
+        float
+        int
+        path
+        str
+        unescapedType
+      ]);
+      valueType' = (either valueType (listOf valueType)) // {
+        description = "Environment variable";
+      };
+    in attrsOf valueType';
+
+    lib = {
+      mkUnescaped = value: {
+        _type = "unescaped";
+        inherit value;
+      };
+    };
+
+    generate = name: value: let
+      valueToString = v: let
```

Comment:
Could you move `valueToString` to `lib/cli.nix`? (and rename it)
It is a pure function (no derivations) so it kind of belongs there, and some contributors would expect to find it there, and may not find this code.

Reply:
Not entirely sure what would be a good name for it. Something like `mkEnvironmentVariable` or `coerceToEnvironmentVariable`?